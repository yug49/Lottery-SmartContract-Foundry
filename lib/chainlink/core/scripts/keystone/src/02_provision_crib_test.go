package src

import (
	"strings"
	"testing"

	"github.com/ethereum/go-ethereum/common"
	"github.com/gkampitakis/go-snaps/snaps"
	"gopkg.in/yaml.v3"
)

func TestGeneratePostprovisionConfig(t *testing.T) {
	chainID := int64(1337)
	capabilitiesP2PPort := int64(6691)
	nodeSetsPath := "./testdata/node_sets.json"
	nodeSetSize := 5
	forwarderAddress := common.Address([20]byte{0: 1}).Hex()
	capabilitiesRegistryAddress := common.Address([20]byte{0: 2}).Hex()
	nodeSets := downloadNodeSets(chainID, nodeSetsPath, nodeSetSize)

	chart := generatePostprovisionConfig(nodeSets, chainID, capabilitiesP2PPort, forwarderAddress, capabilitiesRegistryAddress)

	yamlData, err := yaml.Marshal(chart)
	if err != nil {
		t.Fatalf("Failed to marshal chart: %v", err)
	}

	linesStr := strings.Split(string(yamlData), "\n")
	snaps.MatchSnapshot(t, strings.Join(linesStr, "\n"))
}

func TestGeneratePreprovisionConfig(t *testing.T) {
	nodeSetSize := 5

	chart := generatePreprovisionConfig(nodeSetSize)

	yamlData, err := yaml.Marshal(chart)
	if err != nil {
		t.Fatalf("Failed to marshal chart: %v", err)
	}

	linesStr := strings.Split(string(yamlData), "\n")
	snaps.MatchSnapshot(t, strings.Join(linesStr, "\n"))
}
