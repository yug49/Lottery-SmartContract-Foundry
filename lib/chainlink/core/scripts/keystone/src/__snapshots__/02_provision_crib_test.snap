
[TestGeneratePostprovisionConfig - 1]
helm:
    values:
        chainlink:
            nodes:
                0-ks-wf-bt-node1:
                    image: ${runtime.images.app}
                    overridesToml: |
                        [Capabilities]
                        [Capabilities.Peering]
                        [Capabilities.Peering.V2]
                        Enabled = true
                        ListenAddresses = ['0.0.0.0:6691']

                        [Capabilities.ExternalRegistry]
                        Address = '0x0200000000000000000000000000000000000000'
                        NetworkID = 'evm'
                        ChainID = '1337'

                        [[EVM]]
                        ChainID = '1337'
                        Nodes = []
                0-ks-wf-node2:
                    image: ${runtime.images.app}
                    overridesToml: |
                        [Capabilities]
                        [Capabilities.Peering]
                        [Capabilities.Peering.V2]
                        Enabled = true
                        DefaultBootstrappers = ['12D3KooWFSmZaLFF1nu3mzxPKj43F89WgVDqkpvwFUHBfMHSqpVq@app-0-ks-wf-bt-node1:6691']
                        ListenAddresses = ['0.0.0.0:6691']

                        [Capabilities.ExternalRegistry]
                        Address = '0x0200000000000000000000000000000000000000'
                        NetworkID = 'evm'
                        ChainID = '1337'

                        [[EVM]]
                        ChainID = '1337'
                        Nodes = []

                        [EVM.Workflow]
                        FromAddress = '0x75cf1355cC4Eb358feaBb9e269a4DAEeB6721DBB'
                        ForwarderAddress = '0x0100000000000000000000000000000000000000'
                0-ks-wf-node3:
                    image: ${runtime.images.app}
                    overridesToml: |
                        [Capabilities]
                        [Capabilities.Peering]
                        [Capabilities.Peering.V2]
                        Enabled = true
                        DefaultBootstrappers = ['12D3KooWFSmZaLFF1nu3mzxPKj43F89WgVDqkpvwFUHBfMHSqpVq@app-0-ks-wf-bt-node1:6691']
                        ListenAddresses = ['0.0.0.0:6691']

                        [Capabilities.ExternalRegistry]
                        Address = '0x0200000000000000000000000000000000000000'
                        NetworkID = 'evm'
                        ChainID = '1337'

                        [[EVM]]
                        ChainID = '1337'
                        Nodes = []

                        [EVM.Workflow]
                        FromAddress = '0xc6dcE30f492CBD223b9946603192f22D86e783ca'
                        ForwarderAddress = '0x0100000000000000000000000000000000000000'
                0-ks-wf-node4:
                    image: ${runtime.images.app}
                    overridesToml: |
                        [Capabilities]
                        [Capabilities.Peering]
                        [Capabilities.Peering.V2]
                        Enabled = true
                        DefaultBootstrappers = ['12D3KooWFSmZaLFF1nu3mzxPKj43F89WgVDqkpvwFUHBfMHSqpVq@app-0-ks-wf-bt-node1:6691']
                        ListenAddresses = ['0.0.0.0:6691']

                        [Capabilities.ExternalRegistry]
                        Address = '0x0200000000000000000000000000000000000000'
                        NetworkID = 'evm'
                        ChainID = '1337'

                        [[EVM]]
                        ChainID = '1337'
                        Nodes = []

                        [EVM.Workflow]
                        FromAddress = '0x1289d00A6565Afcd6437B09548F6019EF49696d0'
                        ForwarderAddress = '0x0100000000000000000000000000000000000000'
                0-ks-wf-node5:
                    image: ${runtime.images.app}
                    overridesToml: |
                        [Capabilities]
                        [Capabilities.Peering]
                        [Capabilities.Peering.V2]
                        Enabled = true
                        DefaultBootstrappers = ['12D3KooWFSmZaLFF1nu3mzxPKj43F89WgVDqkpvwFUHBfMHSqpVq@app-0-ks-wf-bt-node1:6691']
                        ListenAddresses = ['0.0.0.0:6691']

                        [Capabilities.ExternalRegistry]
                        Address = '0x0200000000000000000000000000000000000000'
                        NetworkID = 'evm'
                        ChainID = '1337'

                        [[EVM]]
                        ChainID = '1337'
                        Nodes = []

                        [EVM.Workflow]
                        FromAddress = '0x4b92B0aaC39932B7302676F48e78FA91852DC0EE'
                        ForwarderAddress = '0x0100000000000000000000000000000000000000'
                1-ks-str-trig-bt-node1:
                    image: ${runtime.images.app}
                    overridesToml: |
                        [Capabilities]
                        [Capabilities.Peering]
                        [Capabilities.Peering.V2]
                        Enabled = true
                        DefaultBootstrappers = ['12D3KooWFSmZaLFF1nu3mzxPKj43F89WgVDqkpvwFUHBfMHSqpVq@app-0-ks-wf-bt-node1:6691']
                        ListenAddresses = ['0.0.0.0:6691']

                        [Capabilities.ExternalRegistry]
                        Address = '0x0200000000000000000000000000000000000000'
                        NetworkID = 'evm'
                        ChainID = '1337'

                        [[EVM]]
                        ChainID = '1337'
                        Nodes = []
                1-ks-str-trig-node2:
                    image: ${runtime.images.app}
                    overridesToml: |
                        [Capabilities]
                        [Capabilities.Peering]
                        [Capabilities.Peering.V2]
                        Enabled = true
                        DefaultBootstrappers = ['12D3KooWFSmZaLFF1nu3mzxPKj43F89WgVDqkpvwFUHBfMHSqpVq@app-0-ks-wf-bt-node1:6691']
                        ListenAddresses = ['0.0.0.0:6691']

                        [Capabilities.ExternalRegistry]
                        Address = '0x0200000000000000000000000000000000000000'
                        NetworkID = 'evm'
                        ChainID = '1337'

                        [[EVM]]
                        ChainID = '1337'
                        Nodes = []
                1-ks-str-trig-node3:
                    image: ${runtime.images.app}
                    overridesToml: |
                        [Capabilities]
                        [Capabilities.Peering]
                        [Capabilities.Peering.V2]
                        Enabled = true
                        DefaultBootstrappers = ['12D3KooWFSmZaLFF1nu3mzxPKj43F89WgVDqkpvwFUHBfMHSqpVq@app-0-ks-wf-bt-node1:6691']
                        ListenAddresses = ['0.0.0.0:6691']

                        [Capabilities.ExternalRegistry]
                        Address = '0x0200000000000000000000000000000000000000'
                        NetworkID = 'evm'
                        ChainID = '1337'

                        [[EVM]]
                        ChainID = '1337'
                        Nodes = []
                1-ks-str-trig-node4:
                    image: ${runtime.images.app}
                    overridesToml: |
                        [Capabilities]
                        [Capabilities.Peering]
                        [Capabilities.Peering.V2]
                        Enabled = true
                        DefaultBootstrappers = ['12D3KooWFSmZaLFF1nu3mzxPKj43F89WgVDqkpvwFUHBfMHSqpVq@app-0-ks-wf-bt-node1:6691']
                        ListenAddresses = ['0.0.0.0:6691']

                        [Capabilities.ExternalRegistry]
                        Address = '0x0200000000000000000000000000000000000000'
                        NetworkID = 'evm'
                        ChainID = '1337'

                        [[EVM]]
                        ChainID = '1337'
                        Nodes = []
                1-ks-str-trig-node5:
                    image: ${runtime.images.app}
                    overridesToml: |
                        [Capabilities]
                        [Capabilities.Peering]
                        [Capabilities.Peering.V2]
                        Enabled = true
                        DefaultBootstrappers = ['12D3KooWFSmZaLFF1nu3mzxPKj43F89WgVDqkpvwFUHBfMHSqpVq@app-0-ks-wf-bt-node1:6691']
                        ListenAddresses = ['0.0.0.0:6691']

                        [Capabilities.ExternalRegistry]
                        Address = '0x0200000000000000000000000000000000000000'
                        NetworkID = 'evm'
                        ChainID = '1337'

                        [[EVM]]
                        ChainID = '1337'
                        Nodes = []
        ingress:
            hosts:
                - host: ${DEVSPACE_NAMESPACE}-0-ks-wf-bt-node1.${DEVSPACE_INGRESS_BASE_DOMAIN}
                  http:
                    paths:
                        - path: /
                          backend:
                            service:
                                name: app-0-ks-wf-bt-node1
                                port:
                                    number: 6688
                - host: ${DEVSPACE_NAMESPACE}-0-ks-wf-node2.${DEVSPACE_INGRESS_BASE_DOMAIN}
                  http:
                    paths:
                        - path: /
                          backend:
                            service:
                                name: app-0-ks-wf-node2
                                port:
                                    number: 6688
                - host: ${DEVSPACE_NAMESPACE}-0-ks-wf-node3.${DEVSPACE_INGRESS_BASE_DOMAIN}
                  http:
                    paths:
                        - path: /
                          backend:
                            service:
                                name: app-0-ks-wf-node3
                                port:
                                    number: 6688
                - host: ${DEVSPACE_NAMESPACE}-0-ks-wf-node4.${DEVSPACE_INGRESS_BASE_DOMAIN}
                  http:
                    paths:
                        - path: /
                          backend:
                            service:
                                name: app-0-ks-wf-node4
                                port:
                                    number: 6688
                - host: ${DEVSPACE_NAMESPACE}-0-ks-wf-node5.${DEVSPACE_INGRESS_BASE_DOMAIN}
                  http:
                    paths:
                        - path: /
                          backend:
                            service:
                                name: app-0-ks-wf-node5
                                port:
                                    number: 6688
                - host: ${DEVSPACE_NAMESPACE}-1-ks-str-trig-bt-node1.${DEVSPACE_INGRESS_BASE_DOMAIN}
                  http:
                    paths:
                        - path: /
                          backend:
                            service:
                                name: app-1-ks-str-trig-bt-node1
                                port:
                                    number: 6688
                - host: ${DEVSPACE_NAMESPACE}-1-ks-str-trig-node2.${DEVSPACE_INGRESS_BASE_DOMAIN}
                  http:
                    paths:
                        - path: /
                          backend:
                            service:
                                name: app-1-ks-str-trig-node2
                                port:
                                    number: 6688
                - host: ${DEVSPACE_NAMESPACE}-1-ks-str-trig-node3.${DEVSPACE_INGRESS_BASE_DOMAIN}
                  http:
                    paths:
                        - path: /
                          backend:
                            service:
                                name: app-1-ks-str-trig-node3
                                port:
                                    number: 6688
                - host: ${DEVSPACE_NAMESPACE}-1-ks-str-trig-node4.${DEVSPACE_INGRESS_BASE_DOMAIN}
                  http:
                    paths:
                        - path: /
                          backend:
                            service:
                                name: app-1-ks-str-trig-node4
                                port:
                                    number: 6688
                - host: ${DEVSPACE_NAMESPACE}-1-ks-str-trig-node5.${DEVSPACE_INGRESS_BASE_DOMAIN}
                  http:
                    paths:
                        - path: /
                          backend:
                            service:
                                name: app-1-ks-str-trig-node5
                                port:
                                    number: 6688

---

[TestGeneratePreprovisionConfig - 1]
helm:
    values:
        chainlink:
            nodes:
                0-ks-wf-bt-node1:
                    image: ${runtime.images.app}
                0-ks-wf-node2:
                    image: ${runtime.images.app}
                0-ks-wf-node3:
                    image: ${runtime.images.app}
                0-ks-wf-node4:
                    image: ${runtime.images.app}
                0-ks-wf-node5:
                    image: ${runtime.images.app}
                1-ks-str-trig-bt-node1:
                    image: ${runtime.images.app}
                1-ks-str-trig-node2:
                    image: ${runtime.images.app}
                1-ks-str-trig-node3:
                    image: ${runtime.images.app}
                1-ks-str-trig-node4:
                    image: ${runtime.images.app}
                1-ks-str-trig-node5:
                    image: ${runtime.images.app}
        ingress:
            hosts:
                - host: ${DEVSPACE_NAMESPACE}-0-ks-wf-bt-node1.${DEVSPACE_INGRESS_BASE_DOMAIN}
                  http:
                    paths:
                        - path: /
                          backend:
                            service:
                                name: app-0-ks-wf-bt-node1
                                port:
                                    number: 6688
                - host: ${DEVSPACE_NAMESPACE}-0-ks-wf-node2.${DEVSPACE_INGRESS_BASE_DOMAIN}
                  http:
                    paths:
                        - path: /
                          backend:
                            service:
                                name: app-0-ks-wf-node2
                                port:
                                    number: 6688
                - host: ${DEVSPACE_NAMESPACE}-0-ks-wf-node3.${DEVSPACE_INGRESS_BASE_DOMAIN}
                  http:
                    paths:
                        - path: /
                          backend:
                            service:
                                name: app-0-ks-wf-node3
                                port:
                                    number: 6688
                - host: ${DEVSPACE_NAMESPACE}-0-ks-wf-node4.${DEVSPACE_INGRESS_BASE_DOMAIN}
                  http:
                    paths:
                        - path: /
                          backend:
                            service:
                                name: app-0-ks-wf-node4
                                port:
                                    number: 6688
                - host: ${DEVSPACE_NAMESPACE}-0-ks-wf-node5.${DEVSPACE_INGRESS_BASE_DOMAIN}
                  http:
                    paths:
                        - path: /
                          backend:
                            service:
                                name: app-0-ks-wf-node5
                                port:
                                    number: 6688
                - host: ${DEVSPACE_NAMESPACE}-1-ks-str-trig-bt-node1.${DEVSPACE_INGRESS_BASE_DOMAIN}
                  http:
                    paths:
                        - path: /
                          backend:
                            service:
                                name: app-1-ks-str-trig-bt-node1
                                port:
                                    number: 6688
                - host: ${DEVSPACE_NAMESPACE}-1-ks-str-trig-node2.${DEVSPACE_INGRESS_BASE_DOMAIN}
                  http:
                    paths:
                        - path: /
                          backend:
                            service:
                                name: app-1-ks-str-trig-node2
                                port:
                                    number: 6688
                - host: ${DEVSPACE_NAMESPACE}-1-ks-str-trig-node3.${DEVSPACE_INGRESS_BASE_DOMAIN}
                  http:
                    paths:
                        - path: /
                          backend:
                            service:
                                name: app-1-ks-str-trig-node3
                                port:
                                    number: 6688
                - host: ${DEVSPACE_NAMESPACE}-1-ks-str-trig-node4.${DEVSPACE_INGRESS_BASE_DOMAIN}
                  http:
                    paths:
                        - path: /
                          backend:
                            service:
                                name: app-1-ks-str-trig-node4
                                port:
                                    number: 6688
                - host: ${DEVSPACE_NAMESPACE}-1-ks-str-trig-node5.${DEVSPACE_INGRESS_BASE_DOMAIN}
                  http:
                    paths:
                        - path: /
                          backend:
                            service:
                                name: app-1-ks-str-trig-node5
                                port:
                                    number: 6688

---
